### DISCUSSION TITLE
Is GitHub's Markdown renderer open source?

### DISCUSSION THREAD

**Discussion by ugultopu** [community/community#24327] [Apps, API and Webhooks]
<p>GitHub <a href="https://docs.github.com/rest/reference/markdown" rel="noopener nofollow ugc">provides an API</a> that one can POST GitHub Flavored Markdown and obtain the rendered HTML as the response.</p>
<p>I was wondering if the code that does this is open source, so that I use it to convert a GitHub Flavored Markdown to HTML locally.</p>

**Comment by lonr** [comment, marked as answer]
<p>That’s <a href="https://github.com/github/markup" rel="noopener nofollow ugc">Markup library</a></p>
<p>From the API doc of <a href="https://docs.github.com/en/rest/overview/media-types#html-1" rel="noopener nofollow ugc">media types</a></p>
<blockquote>
<p>Markup languages are rendered to HTML using our open-source <a href="https://github.com/github/markup" rel="noopener nofollow ugc">Markup library</a>.</p>
</blockquote>
